src/vm/methodtable.cpp

```diff
@@ -9955,6 +9947,35 @@ NOINLINE BYTE *MethodTable::GetLoaderAllocatorObjectForGC()
     return retVal;
 }
 
+UINT32 MethodTable::GetNativeSize()
+{
+    LIMITED_METHOD_CONTRACT;
```

Comment:
These methods should not be `LIMITED_METHOD_CONTRACT`. They do a lot of work inside `EnsureNativeLayoutInfoInitialized`. 

Ideally, these would be `STANDARD_VM_CONTRACT`. If it is too hard to make the callers switch to preemptive mode, these should be `THROWS; MODE_ANY`.